pkg/kncloudevents/event_dispatcher.go

```diff
@@ -339,7 +339,7 @@ func (d *Dispatcher) executeRequest(ctx context.Context, target duckv1.Addressab
 	if isFailure(response.StatusCode) {
 		// Read response body into dispatchInfo for failures
 		if readErr != nil && readErr != io.EOF {
-			dispatchInfo.ResponseBody = []byte(fmt.Sprintf("dispatch resulted in status \"%s\". Could not read response body: error: %s", response.Status, err.Error()))
+			dispatchInfo.ResponseBody = []byte(fmt.Sprintf("dispatch resulted in status \"%s\". Could not read response body: error: %s", response.Status, readErr.Error()))
```

Comment:
is there a reason we don't just use `err` ?

Reply:
I think this was mostly copied over from the old dispatcher version IIRC. I can update it...